test/checksum.cpp

```diff
@@ -20,56 +19,46 @@ unsigned long get_size_by_fd(int fd) {
 }
 
 // Function which compares checksums of the files passed
-void compare_hash(char* filename_1, char* filename_2) 
-{
+void compare_hash(char* filename_1, char* filename_2) {
     int file_descript_1;
     int file_descript_2;
     unsigned long file_size_1,file_size_2;
     char* file_buffer_1;
     char* file_buffer_2;
 
-    if(!filename_1 || !filename_2) { 
-            printf("Must specify two files to be compared\n");
-            exit(-1);
+    if(!filename_1 || !filename_2) {
+        printf("Must specify two files to be compared\n");
+        exit(-1);
     }
     printf("Orignal file:\t%s\n", filename_1);
     printf("File to be checked:\t%s\n", filename_2);
 
     file_descript_1 = open(filename_1, O_RDONLY);
     if(file_descript_1 < 0) exit(-1);
-	
-	file_descript_2 = open(filename_2, O_RDONLY);
+
+    file_descript_2 = open(filename_2, O_RDONLY);
     if(file_descript_2 < 0) exit(-1);
-	
+
     file_size_1 = get_size_by_fd(file_descript_1);
-   
-	file_size_2 = get_size_by_fd(file_descript_2);
- 
-	
+
+    file_size_2 = get_size_by_fd(file_descript_2);
 
     file_buffer_1 = static_cast<char*>(mmap(0, file_size_1, PROT_READ, MAP_SHARED, file_descript_1, 0));
     result_1 = g_compute_checksum_for_string(G_CHECKSUM_MD5,file_buffer_1,file_size_1);
-	printf("%s(checksum):",filename_1);
+    printf("%s(checksum):",filename_1);
     printf("%s\n",result_1);
-    
+
     file_buffer_2 = static_cast<char*>(mmap(0, file_size_2, PROT_READ, MAP_SHARED, file_descript_2, 0));
     result_2 = g_compute_checksum_for_string(G_CHECKSUM_MD5,file_buffer_2,file_size_2);
-	printf("%s(checksum):",filename_2);
+    printf("%s(checksum):",filename_2);
     printf("%s\n",result_2);
-   
-    if(strcmp(reinterpret_cast<char*>(result_1),reinterpret_cast<char*>(result_2))==0)
-    {
-		printf("Data Integrity Test Passed...MD5 Checksum is Same for Both Files\n");
-	}
-	
-	else
-	{
-		printf("Data Integrity Test Failed...MD5 Checksum is Not Same for Both Files\n");
-	}
-	
-	g_free(result_1);
-	g_free(result_2);
-    
-}
 
- 
+    if(strcmp(reinterpret_cast<char*>(result_1),reinterpret_cast<char*>(result_2))==0) {
```

Comment:
I wasn't sure if this was only a C++ convention.  Along the same lines, it would probably be safer to use static_cast<type> instead of the regular cast (type) in the various places we do that.